lms/djangoapps/user_tours/management/commands/backpopulate_user_tours.py

```diff
@@ -0,0 +1,40 @@
+""" Management command to backpopulate User Tours. """
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+
+from common.djangoapps.student.models import CourseEnrollment
+from lms.djangoapps.user_tours.models import UserTour
+
+User = get_user_model()
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms backpopulate_user_tours
+    """
+    help = 'Creates or updates a row in the UserTour table for all users in the platform.'
+
+    def handle(self, *args, **options):
+        """
+        Backpopulates UserTour objects for all existing users.
+
+        If the user has any prior enrollments, we will treat them as an existing user,
+        otherwise they will receive a new user treatment.
+        """
+        for user in User.objects.all():
```

Comment:
Do we want to segment this at all via arguments given in the database or something? This seems like a long task potentially? But I guess it's only locking the UserTour table during that time?

Reply:
Yeah, I wouldn't think it would be locking out the User table. It might be decently long running. We can maybe try running it in stage and if we are concerned, updating the script to handle segmenting or something?

One potential speedup that I didn't build in because it doesn't really apply in a normal situation is to filter out all users who already have UserTours created. That becomes more applicable if we need to rerun the job multiple times, but first time through, wouldn't help at all.

Reply:
If we do rerun this job, it will reset any users that have seen the tour already. Yeah, maybe change the for loop to `Users.objects.filter(user_tour=None)`? (I forget if that is right syntax, or you need `__is_null=True`)

Reply:
Yeah, I guess defensive is better. I didn't do it originally because I figured in practice, this job will be run prior to enabling any tours so you'd be safe in rerunning it because they couldn't have seen any tours yet. Will update it.

Reply:
Added defensive code to ignore anyone with a UserTour